src/System.Reflection.Metadata/src/System/Reflection/Internal/MemoryBlocks/AbstractMemoryBlock.cs

```diff
@@ -4,13 +4,14 @@
 
 using System.Collections.Immutable;
 using System.Reflection.Metadata;
+using System.Runtime.ConstrainedExecution;
 
 namespace System.Reflection.Internal
 {
     /// <summary>
     /// Represents a disposable blob of memory accessed via unsafe pointer.
     /// </summary>
-    internal abstract class AbstractMemoryBlock : IDisposable
+    internal abstract class AbstractMemoryBlock : CriticalFinalizerObject, IDisposable
```

Comment:
I do not think there is alternative. However, you may need to add netfx specific build flavor to make sure that this kicks in for .NET Framework.